cli/CmdRunner.go

```diff
@@ -69,8 +75,62 @@ func (cc *CommandContext) Err() error {
 }
 
 type CmdRunner struct {
-	sim *simulation.Simulation
-	ctx *progctx.ProgCtx
+	sim           *simulation.Simulation
+	ctx           *progctx.ProgCtx
+	contextNodeId NodeId
+}
+
+func (rt *CmdRunner) HandleCommand(cmdline string, output io.Writer) error {
+	if rt.contextNodeId != InvalidNodeId && !isContextlessCommand(cmdline) {
+		// run the command in node context
+		cmd := &Command{
+			Node: &NodeCmd{
+				Node:    NodeSelector{Id: rt.contextNodeId},
+				Command: &cmdline,
+			},
+		}
+		cc := rt.Execute(cmd)
+
+		if cc.Err() != nil {
+			if _, err := fmt.Fprintf(output, "Error: %v\n", cc.Err()); err != nil {
+				return err
+			}
+		} else {
+			if _, err := fmt.Fprintf(output, "Done\n"); err != nil {
+				return err
+			}
+		}
+	} else {
+		// run the OTNS-CLI command
+		cmd := &Command{}
+		if err := ParseBytes([]byte(cmdline), cmd); err != nil {
+			if _, err := fmt.Fprintf(os.Stdout, "Error: %v\n", err); err != nil {
+				return err
+			}
+		} else {
+			cc := rt.Execute(cmd)
+
+			if cc.Err() != nil {
+				if _, err := fmt.Fprintf(os.Stdout, "Error: %v\n", cc.Err()); err != nil {
```

Comment:
nit: write short functions to print with consistent formats?

Reply:
good suggestion, but unrelated to this PR. Maybe later when we have a better chance?